### PR TITLE
Q7

### PR DESCRIPTION
**Pull request by Byteuth**
Adds CORE: POST /api/articles/:article_id/comments
Adds tests for new POST
Adds new controllers, models functions
Adds testing for new models

Adds some errorhandling

### CODE REVIEW

File: __tests__/utils.models.test.js [hunk 1]
```diff
@@ -50,4 +56,35 @@ describe("selectCommentsById", () => {
 
 })
 
+describe("insertComment", () => {
+    test('tests selectCommentsById for correct properties, tests to see if comments db increase with each post ', async () => {
```
Comment (LangaLee) [inline __tests__/utils.models.test.js:60]: Lets not mix async await with promise chains, choose 1 😄

File: controllers/articles.controllers.js [hunk 2]
```diff
@@ -33,4 +34,15 @@ exports.getCommentsByArticleId = (req, res, next) => {
...
     .catch(next)
+}
+
+
+exports.addCommentToArticleId = (req, res, next) => {
+    const comment = req.body
+    const articleId = req.params.article_id
+    insertComment(articleId, comment)
+    .then((postedComment) => {
+        res.status(200).send({postedComment: postedComment});
```
Comment (LangaLee) [inline controllers/articles.controllers.js:45]: Should we send a 200 ? have a read on 200, 201, 204 on http cats or the notes

File: __tests__/app.test.js [hunk 3]
```diff
@@ -161,6 +161,45 @@ describe("/api/articles/:article_id/comments", () => {
...
+
+            expect(comment.hasOwnProperty('comment_id')).toBe(true)
+            expect(comment.hasOwnProperty('article_id')).toBe(true)
+            expect(comment.hasOwnProperty('votes')).toBe(true)
+            expect(comment.hasOwnProperty('created_at')).toBe(true)
+
+        });
+    });
+
+    test("POST:422 sends an appropriate status and error message when given a valid but non-existent id", () => {
```
Comment (LangaLee) [inline __tests__/app.test.js:189]: Would this not be a 400 😄

File: __tests__/utils.models.test.js [hunk 4]
```diff
@@ -50,4 +56,35 @@ describe("selectCommentsById", () => {
 
 })
 
+describe("insertComment", () => {
```
Comment (LangaLee) [inline __tests__/utils.models.test.js:59]: Do we need this file to test models? In the app test we are testing endpoints which do send back data and make use of the models.

**Review by LangaLee — COMMENTED** [review]
left some feedback below 😄

File: controllers/articles.controllers.js (hunk 2 above)
Comment (Byteuth) [reply controllers/articles.controllers.js:45]: +

File: __tests__/app.test.js (hunk 3 above)
Comment (Byteuth) [reply __tests__/app.test.js:189]: +

File: __tests__/utils.models.test.js (hunk 4 above)
Comment (Byteuth) [reply __tests__/utils.models.test.js:59]: good point!
should i count the increments of each comment_id in app.test instead?

**Pull request merged**